libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -307,6 +313,62 @@ public void rerouteOccurred() {
     queuedRerouteEvents.add(mapboxNavigation.getSessionState());
   }
 
+  public String recordFeedbackEvent(String feedbackType, String description,
+                                    @FeedbackEvent.FeedbackSource String feedbackSource) {
+
+    // Get current session state and update with "before" locations (equal to current state of the location buffer)
+    SessionState feedbackEventSessionState = mapboxNavigation.getSessionState().toBuilder()
+      .beforeRerouteLocations(Arrays.asList(
+        locationBuffer.toArray(new Location[locationBuffer.size()])))
+      .routeProgressBeforeReroute(routeProgress)
+      .build();
+
+    FeedbackEvent feedbackEvent = new FeedbackEvent(feedbackEventSessionState, feedbackSource);
+    // TODO set feedback type and description from the parameters based
+    // TODO - note: these should use default values if the values passed are empty
```

Comment:
Default for `feedbackType` is `FEEDBACK_TYPE_GENERAL` and `description` is just empty String.